### PR TITLE
Feature/satcom merge

### PR DESCRIPTION
**Pull request by mhugentobler**
- Satcom functionality
--> mode switching between satcom and telemetry operation
--> handle asl_high_latency
--> making sure stuff (params, mission, commands etc) is sent on correct link (satcom vs. telemetry)
--> param autosave feature

- ASL widgets adapted to new qgc version
--> asl_high_latency message updates energy widget during satcom operation

### CODE REVIEW

File: src/ui/preferences/LinkSettings.qml [hunk 1]
```diff
@@ -271,21 +275,21 @@ Rectangle {
...
+//                        onCheckedChanged: {
+//                            if(editConfig) {
+//                                editConfig.autoConnect = checked
+//                            }
+//                        }
+//                        Component.onCompleted: {
+//                            if(editConfig)
+//                                checked = editConfig.autoConnect
+//                        }
+//                    }
```
Comment (tstastny) [inline src/ui/preferences/LinkSettings.qml]: Is this the auto connect feature that also works with the telemetry? I actually liked this... it can be disabled with the checkbox, right? So probably should not comment the source code out! :S Or does it interfere with other things?

File: src/uas/UAS.h [hunk 2]
```diff
@@ -357,7 +357,6 @@ public slots:
     bool componentMulti[256];
     bool connectionLost; ///< Flag indicates a timed out connection
     quint64 connectionLossTime; ///< Time the connection was interrupted
-    quint64 lastVoltageWarning; ///< Time at which the last voltage warning occurred
```
Comment (tstastny) [inline src/uas/UAS.h:360]: Was this simply not used? Why did you delete?

**Review by tstastny — COMMENTED** [review]
I cant judge too well on most things -- you went quite deep into QGC! Only two comments. Let me know answers to those, then otherwise should be fine to merge.

File: src/ui/preferences/LinkSettings.qml (hunk 1 above)
Comment (mhugentobler) [reply src/ui/preferences/LinkSettings.qml]: this shouldnt interfere with anything i made. i think i commented this because i didnt want the autoconnect for some testing cases and forgot i left it commented

File: src/uas/UAS.h (hunk 2 above)
Comment (mhugentobler) [reply src/uas/UAS.h:360]: the uas class is deprecated. i put this lastvoltagewarning in the vehicle class.
originally our widgets were based on the asluav class which inherits from uas. since uas is deprecated i decided to just put everything into vehicle

File: src/ui/preferences/LinkSettings.qml (hunk 1 above)
Comment (tstastny) [reply src/ui/preferences/LinkSettings.qml]: ok - I would put this back in then ;)

File: src/uas/UAS.h (hunk 2 above)
Comment (tstastny) [reply src/uas/UAS.h:360]: ok

**Review by tstastny — APPROVED** [review]
